samwell/sam/sambuilder.py

```diff
@@ -114,14 +117,28 @@ def __init__(self,
             extra_header: a dictionary of extra values to add to the header, None otherwise.  See
                           `::class::~pysam.AlignmentHeader` for more details.
             seed: a seed value for random number/string generation
+            sort_order: optional sort order, if none reads will be output in the same order as they
+                were appended. If `SamOrder.Coordinate`, reads will be ordered by reference index
+                and coordinate order. If `SamOrder.QueryName`, reads will be ordered by query name.
         """
 
         self.r1_len: int = r1_len if r1_len is not None else self.DEFAULT_R1_LENGTH
         self.r2_len: int = r2_len if r2_len is not None else self.DEFAULT_R2_LENGTH
         self.base_quality: int = base_quality
         self.mapping_quality: int = mapping_quality
+
+        self.sort_order: SamOrder = sort_order
+        if sort_order is None:
+            sort_order_header = "unsorted"
+        elif sort_order == SamOrder.Coordinate:
+            sort_order_header = "coordinate"
+        elif sort_order == SamOrder.QueryName:
+            sort_order_header = "queryname"
+        else:
+            raise ValueError("sort_order provided must be one of Coordinate, QueryName, or None")
```

Comment:
1. Make the `SamOrder` enum have an attribute that gives the name in the SAM header.
2. Add an `Unsorted` enum.
3. `self.sort_order: SamOrder = sort_order` becomes `self.sort_order: SamOrder = SamOrder.Unsorted if sort_order is None else sort_order`